Перебор веб страниц

pankratsdarya opened the pull request:
Задание:
1. Доработать программу из практической части так, чтобы при отправке ей сигнала SIGUSR1 она увеличивала глубину поиска на 10.
2. Добавить общий таймаут на выполнение следующих операций: работа парсера, получений ссылок со страницы, формирование заголовка.

Реализация:
В задании 1 используется сигнал SIGINT, т.к. в винде нет SIGUSR1.

pankratsdarya committed:
Перебор веб страниц

Задание:
1. Доработать программу из практической части так, чтобы при отправке ей сигнала SIGUSR1 она увеличивала глубину поиска на 10.
2. Добавить общий таймаут на выполнение следующих операций: работа парсера, получений ссылок со страницы, формирование заголовка.

Реализация:
В задании 1 используется сигнал SIGINT, т.к. в винде нет SIGUSR1.

kilchik commented on crawler/crawler.go:
```diff
@@ -0,0 +1,90 @@
+			return
+		}
+
+		page, err := parse(url, ctxTime)
```
не ошибка, но идиоматично передавать ctx первым аргументом

kilchik commented on crawler/ugly_parser.go:
```diff
@@ -0,0 +1,80 @@
+
+	default:
+		// что здесь должно быть вместо http.Get? :)
+		r, err := http.Get(url)
```
как долго может висеть http.Get? сработает ли здесь таймаут?

kilchik commented on crawler/main.go:
```diff
@@ -0,0 +1,129 @@
+	log.Printf("got signal %q", sig.String())
+
+	// если сигнал получен, увеличиваем maxDepth
+	craw.maxDepth++
```
что будет при попытке одновременного чтения и записи maxDepth? эти операции проводятся в разных горутинах

pankratsdarya committed:
Hotfix

1. Поправлен порядок передачи аргументов - контекст везде передается первым.
2.  Изменение глубины обложено мьютексом.

pankratsdarya committed:
Hotfix 2

1. Замена http.Get() конструкцией с таймаутом.

pankratsdarya committed:
Hotfix 3

Увеличено количество мьютексов. maxDepth обложен мьютексами при всех обращениях.
